### PR TITLE
Use RemoteCredProvider instead of EC2RoleProvider 

### PR DESCRIPTION
**Pull request by jefferai**
This better handles ECS and other future use-cases.

### CODE REVIEW

**Comment by jefferai** [comment]
@joelthompson It's not letting me request a review from you but can you take a look?

**Comment by joelthompson** [comment]
@jefferai -- this generally seems fine (modulo the compile error from Travis), but I don't have too much experience with the golang SDK. I would suggest updating the documentation to reflect these changes, i.e.:

* In the [AWS secret backend](https://www.vaultproject.io/docs/secrets/aws/index.html), for the `region` option, mention the environment variable, and updating "Note: the client uses the official AWS SDK and will use environment variable or IAM role-provided credentials if available." to also add ECS task role.
* In the [AWS auth backend](https://www.vaultproject.io/docs/auth/aws.html), where it states, "environment variables, ~/.aws/credentials, IAM instance profile in that order" to add ECS task role
* The [S3 storage backend](https://www.vaultproject.io/docs/configuration/storage/s3.html), for the region option, note that it'll use the `AWS_REGION` environment variable if `AWS_DEFAULT_REGION` isn't set (and if you're in there, fix the `endpoint` parameter doc as it has the wrong environment variable)

There's some more cleanup work that could be done to make things more consistent, e.g., move the DynamoDB physical backend to use the credential helper, as it won't pick up creds from ECS tasks, be consistent about using AWS_REGION and AWS_DEFAULT_REGION environment variables, and make AWS_REGION take precedence over AWS_DEFAULT_REGION which is what the golang SDK does -- though that's a small a breaking change. But, I'd suggest just getting this fix in now for the next release, and potentially do the cleanup work later.

**Comment by jefferai** [comment]
> be consistent about using AWS_REGION and AWS_DEFAULT_REGION environment variables, and make AWS_REGION take precedence over AWS_DEFAULT_REGION which is what the golang SDK does

I didn't even see AWS_DEFAULT_REGION in the sdk, I thought that was something Vault-specific.

Anyways, I've updated with your feedback, except for updating the docs. Can you take another look?

File: builtin/logical/aws/client.go [hunk 1]
```diff
@@ -31,7 +32,10 @@ func getRootConfig(s logical.Storage) (*aws.Config, error) {
 	}
 
 	if credsConfig.Region == "" {
-		credsConfig.Region = "us-east-1"
+		credsConfig.Region = os.Getenv("AWS_REGION")
```
Comment (joelthompson) [inline builtin/logical/aws/client.go]: Do you also want to check AWS_DEFAULT_REGION here?

Comment (jefferai) [inline builtin/logical/aws/client.go]: Done!

File: website/source/docs/configuration/storage/s3.html.md [hunk 2]
```diff
@@ -35,10 +35,11 @@ storage "s3" {
 
 - `endpoint` `(string: "")` – Specifies an alternative, AWS compatible, S3
   endpoint. This can also be provided via the environment variable
-  `AWS_DEFAULT_REGION`.
+  `AWS_S3_ENDPOINT`.
```
Comment (briankassouf) [inline website/source/docs/configuration/storage/s3.html.md]: Was this incorrect before?

Comment (jefferai) [inline website/source/docs/configuration/storage/s3.html.md]: Yep. Looks like it was copypasta. The code actually uses AWS_S3_ENDPOINT though.

**Pull request merged**